### PR TITLE
add notebooknodeexplorer shortcut copy paste

### PR DESCRIPTION
**Pull request by feloxx**
Add notebooknodeexplorer global shortcut

In `NotebookNodeExplorer::setupShortcuts` impl

- Copy
- Paste

---

But I don't know what the second false means. `copySelectedNodes`

### CODE REVIEW

File: src/widgets/notebooknodeexplorer.cpp [hunk 1]
```diff
@@ -2182,6 +2182,29 @@ void NotebookNodeExplorer::setupShortcuts()
...
 
+    // Copy
+    {
+        auto shortcut = WidgetUtils::createShortcut(coreConfig.getShortcut(CoreConfig::Copy), this);
+        if (shortcut) {
+            connect(shortcut, &QShortcut::activated,
+                    this, [this]() {
+                        copySelectedNodes(false, false);
+                    });
+
```
Comment (tamlok) [inline src/widgets/notebooknodeexplorer.cpp:2193]: Remove this empty line.

File: src/widgets/notebooknodeexplorer.cpp [hunk 2]
```diff
@@ -2182,6 +2182,29 @@ void NotebookNodeExplorer::setupShortcuts()
...
+    }
+
+    // Paste
+    {
+        auto shortcut = WidgetUtils::createShortcut(coreConfig.getShortcut(CoreConfig::Paste), this);
+        if (shortcut) {
+            connect(shortcut, &QShortcut::activated,
+                    this, [this]() {
+                        pasteNodesFromClipboard();
+                    });
```
Comment (tamlok) [inline src/widgets/notebooknodeexplorer.cpp:2204]: You can write this line as line 1194.

File: src/widgets/notebooknodeexplorer.cpp [hunk 3]
```diff
@@ -2182,6 +2182,29 @@ void NotebookNodeExplorer::setupShortcuts()
         }
     }
 
+    // Copy
+    {
+        auto shortcut = WidgetUtils::createShortcut(coreConfig.getShortcut(CoreConfig::Copy), this);
+        if (shortcut) {
+            connect(shortcut, &QShortcut::activated,
+                    this, [this]() {
+                        copySelectedNodes(false, false);
```
Comment (tamlok) [inline src/widgets/notebooknodeexplorer.cpp:2191]: Get the second parameter like line 2176~2179.

It means whether it is from the master explorer or the slaver explorer when in separate view mode.

**Review by tamlok — CHANGES_REQUESTED** [review]

File: src/widgets/notebooknodeexplorer.cpp (hunk 1 above)
Comment (feloxx) [reply src/widgets/notebooknodeexplorer.cpp:2193]: done

File: src/widgets/notebooknodeexplorer.cpp (hunk 3 above)
Comment (feloxx) [reply src/widgets/notebooknodeexplorer.cpp:2191]: done

File: src/widgets/notebooknodeexplorer.cpp (hunk 2 above)
Comment (feloxx) [reply src/widgets/notebooknodeexplorer.cpp]: done

File: src/widgets/notebooknodeexplorer.cpp [hunk 4]
```diff
@@ -2163,6 +2163,15 @@ void NotebookNodeExplorer::addOpenWithMenu(QMenu *p_menu, bool p_master)
                        });
 }
 
+// Shortcut auxiliary, it can also be used to determine the browser.
+bool NotebookNodeExplorer::isActionFromMaster() const
+{
+    if (!isCombinedExploreMode()) {
+        return m_masterExplorer->hasFocus();
+    }
+    return false;
```
Comment (tamlok) [inline src/widgets/notebooknodeexplorer.cpp:2172]: return true by default?

**Review by tamlok — CHANGES_REQUESTED** [review]

File: src/widgets/notebooknodeexplorer.cpp (hunk 4 above)
Comment (feloxx) [reply src/widgets/notebooknodeexplorer.cpp]: done

**Pull request merged**